homeassistant/components/unifi_access/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for the UniFi Access integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import uiaccessclient
+import urllib3.exceptions
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_API_TOKEN, CONF_HOST, CONF_VERIFY_SSL
+from homeassistant.core import HomeAssistant
+
+from .const import DEFAULT_HOST, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_API_TOKEN): str,
+        vol.Required(CONF_VERIFY_SSL): bool,
+    }
+)
+
+
+class UniFiAccessConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for UniFi Access."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Process configuration form."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            info = await _validate_input(self.hass, user_input, errors)
+            if not errors:
+                return self.async_create_entry(title=info["title"], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+
+async def _validate_input(
+    hass: HomeAssistant, data: dict[str, Any], errors: dict[str, str]
+) -> dict[str, Any]:
+    host = f"https://{data[CONF_HOST]}/api/v1/developer"
+    configuration = uiaccessclient.Configuration(
+        host, access_token=data[CONF_API_TOKEN]
+    )
+    configuration.verify_ssl = data[CONF_VERIFY_SSL]
+    api_client = uiaccessclient.ApiClient(configuration)
+    space_api = uiaccessclient.SpaceApi(api_client)
+
+    try:
+        await hass.async_add_executor_job(space_api.fetch_all_doors)
+    except (
+        uiaccessclient.exceptions.UnauthorizedException,
+        uiaccessclient.exceptions.ForbiddenException,
+    ):
+        errors["base"] = "invalid_auth"
+    except urllib3.exceptions.HTTPError:
+        errors["base"] = "cannot_connect"
+    except Exception:
+        _LOGGER.exception("Unexpected exception")
+        errors["base"] = "unknown"
+
+    return {"title": "UniFi Access"}
```

Comment:
Returning a static title doesnt really do much more than returning nothing

Reply:
Agreed. This is from the auto generated template from Home Assistant. Would it be preferable to hard code this where it is used, or is there another way of setting static titles? Or possible to include the hostname or similar in the title to make entries unique?

Reply:
I've hard-coded this in the `self.async_create_entry()` call instead, please advice if it would be better to use the hostname in the title to distinguish between entries or if this is preferable as is.